File: _en/Computer/Operating_System/C&C++_Standard_Library/foreach.cpp
/* foreach.cpp
Author: BSS9395
Update: 2021-10-11T22:24:00+08@China-Guangdong-Shenzhen+08
Design: C++ foreach Loop
*/

#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main(int argc, char *argv[]) {
    vector<string> strs;
    strs.push_back("C");
    strs.push_back("C++");
    strs.push_back("C#");

    for (string str : strs) {
        str = "Do";           // Original value will not change.
        cout << str << endl;
    }

    cout << "========================================" << endl;
    for (string &str : strs) {
        cout << str << endl;
        str = "Do";          // Original value will change.
    }

    cout << "========================================" << endl;
    for (auto &str : strs) {
        cout << str << endl;
    }

    return 0;
}

File: _en/Computer/Operating_System/Java_Standard_Library/foreach.java
/* foreach.java
Author: BSS9395
Update: 2021-10-10T21:31:00+08@China-Guangdong-Shenzhen+08
Design: Java foreach Loop
*/

public class foreach {
    public static void main(String[] args) {
        String[] strings = new String[]{
                "Java SE",
                "Java EE",
                "Java ME"
        };

        for (String string : strings) {
            string = "Java";                 // Original value will not change.
            System.out.println(string);
        }

        System.out.println("========================================");
        for (int i = 0; i < strings.length; i += 1) {
            System.out.println(strings[i]);
            strings[i] = "Java";            // Original value will change.
        }

        System.out.println("========================================");
        for (var string : strings) {
            System.out.println(string);
        }
    }
}
